### DISCUSSION TITLE
part-ii/2007-23/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#221] [Question comments]
Part II 2007 Coding and Cryptography: 4.I.4G 

https://questions.tripos.org/part-ii/2007-23/

**Comment by walrus1729** [comment]
<details>
<summary>Solution</summary>

The first part is bookwork. I won't type out the full working for Berlekamp-Massey, but I will point out that the matrix

$$A_2 = \begin{pmatrix} 1 & 0 & 1 \\ 0 & 1 & 0 \\ 1 & 0 & 1 \\ \end{pmatrix}$$

has discriminant equal to $0$, but the suggested LFSR $x_n = x_{n-2}$ fails for the later terms of the output.

In fact, we get the solution

$$x_n = x_{n-2} + x_{n-4} + x_{n-5}$$

</details>
